platforms/core-runtime/INSTRUMENTATION.md

```diff
@@ -0,0 +1,437 @@
+# Instrumentation and bytecode upgrades infrastructure
```

Comment:
:x: I think "instrumentation" is used confusingly for multiple purposes, and it would be important to use distinct names for the different aspects here. Here's a proposal:

- what CC does is called "bytecode instrumentation"
- what we do for provider API migration is "bytecode upgrades"
- the infrastructure serving these both is "bytecode interception"

WDYT?

Reply:
I am ok with first two, but 3rd is similar to instrumentation.

What about Bytecode transformation for the infrastructure. So "bytecode transformation" infrastructure.

Reply:
Hmm, or maybe we can go with interception. Let's try it 🤔

Reply:
I now use "interception" for general term when talking about bytecode replacements. And I use "transform" when I am talking about transforming jars or classpath or classes.